src/PCLExt.FileStorage.UWP/StorageFolderImplementation.cs

```diff
@@ -0,0 +1,285 @@
+﻿using PCLExt.FileStorage.UWP.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Storage;
+using Windows.Storage.Search;
+
+namespace PCLExt.FileStorage.UWP
+{
+    sealed class StorageFolderImplementation : IFolder
+    {
+        public string Name => _storageFolder.Name;
+
+        public string Path => _storageFolder.Path;
+
+        public bool Exists => true;
+
+        private readonly StorageFolder _storageFolder;
+
+        public StorageFolderImplementation(StorageFolder storageFolder)
+        {
+            _storageFolder = storageFolder;
+        }
+
+        public StorageFolderImplementation(String path)
+        {
+            _storageFolder = StorageFolder.GetFolderFromPathAsync(path).GetResults();
+        }
+
+        public ExistenceCheckResult CheckExists(string name)
+        {
+            return CheckExistsAsync(name).Result;
```

Comment:
Calling .Result can lead to a deadlock. 
Calling everywhere async code in sync methods is not the best idea guess

Reply:
What do you propose to use for sync method if there is no sync interface for StorageFolder & StorageFile?

Reply:
Yea, that's a bit of a problem

Reply:
Let's decide what should I do and I fix it. My propose is to call async method with .Result. I tested it and it works. If you have another opinion let's discuss it and I'm ready to change my mind.

Reply:
I have no idea for now. Do as you wish. I suggest you notify me when you are finished and I could try to help you with sync in your branch

Reply:
I see only 2 ways here:
1. Call async method inside and return result. +: it will work for user. -: it could be a cause of deadlocks.
2. Throw an NotSupportedException. +: it will show user that call Async method is the only best way. -: user might not expect it

What way do you prefer?
May be another one

Reply:
The problem with calling async inside sync is that we will execute a task each time, what is not good. but it's "acceptable" for now, so it can at least work